CodeGen/Helpers/PrefixBuilder/UnitPrefixBuilder.cs

```diff
@@ -0,0 +1,334 @@
+    ///                 <code>
+    ///                     Requested prefix: Milli (scale -3) for pressure unit Pascal
+    ///                     SI base units of Pascal: L=Meter, M=Kilogram, T=Second
+    ///                     SI base dimensions, ordered: M=1, L=-1, T=-2
+    ///                     Trying first dimension M=1:
+    ///                         SI base unit in mass dimension is Kilogram, but configured base unit is Gram so base prefix scale is 3
+    ///                         ❌ No inferred prefix: base prefix scale 3 + requested prefix scale (-3) = 0
+    ///                     Trying second dimension L=-1:
+    ///                         SI base unit in length dimension is Meter, same as configured base unit, so base prefix scale is 0
+    ///                         Inferred prefix is Milli: base prefix scale 0 + requested prefix scale (-3) = -3
+    ///                         ✅ Resulting base units: M=Millimeter plus the original M=Kilogram, T=Second
+    ///                 </code>
+    ///             </description>
+    ///         </item>
+    ///         <item>
+    ///             <term>Example 3 - ElectricApparentPower.Kilovoltampere</term>
+    ///             <description>
+    ///                 <para>
+    ///                     This example demonstrates cases where determining the base units for certain prefixes is not
+    ///                     possible or trivial.
+    ///                 </para>
+    ///                 <code>
+    ///                     Requested prefix: Kilo (scale 3) for unit Voltampere
+    ///                     SI base units of Voltampere: L=Meter, M=Kilogram, T=Second
+    ///                     SI base dimensions, ordered: M=1, L=2, T=-3
+    ///                     Trying first dimension M=1:
+    ///                         SI base unit in mass dimension is Kilogram, same as configured base unit, so base prefix scale is 0
+    ///                         Inferred prefix is Kilo: base prefix scale 0 + requested prefix scale (3) = 3
+    ///                         ❌ Kilo prefix for Kilogram unit would be Megagram, but there is no unit Megagram, since Gram does not have this prefix (we could add it)
+    ///                     Trying second dimension L=2:
+    ///                         ❌ There is no metric prefix we can raise to the power of 2 and get Kilo, e.g., Deca*Deca = Hecto, Kilo*Kilo = Mega, etc.
+    ///                     Trying third dimension T=-3:
+    ///                         SI base unit in time dimension is Second, same as configured base unit, so base prefix scale is 0
+    ///                         Inferred prefix is Deci: (base prefix scale 0 + requested prefix scale (-3)) / exponent -3 = -3 / -3 = 1
+    ///                         ❌ There is no Duration unit Decasecond (we could add it)
+    ///                 </code>
+    ///             </description>
+    ///         </item>
+    ///     </list>
+    /// </example>
```

Comment:
Poor soul of the next guy trying to digest this, but I can't wait for the bots to pick up my made up words for these concepts 🤣